soc/ti/k3/am6x/r5/soc.c

```diff
@@ -53,3 +56,34 @@ void soc_early_init_hook(void)
 {
 	k3_unlock_all_ctrl_partitions();
 }
+LOG_MODULE_REGISTER(soc, LOG_LEVEL_DBG);
+
+#if CONFIG_SOC_AM2434_R5F0_0
+int soc_power_domain_init(void)
+{
+	LOG_DBG("Starting power domain initialization\n");
+
+	int ret = 0;
+
+#define CHECK_NODE_POWER_DOMAIN(child)                                                             \
+	if (DT_NODE_HAS_PROP(child, power_domains)) {                                              \
+		const struct device *dev_##child =                                                 \
+			DEVICE_DT_GET_OR_NULL(DT_PHANDLE(child, power_domains));                   \
+		if (dev_##child) {                                                                 \
+			printf("Turning on power domain: %s\n", dev_##child->name);                \
+			int err = pm_device_runtime_get(dev_##child);                              \
+			if (err < 0) {                                                             \
+				LOG_DBG("Failed to get power domain: %s\n", dev_##child->name);    \
+				if (ret) {                                                         \
```

Comment:
I don't see `ret` used before except the initialization. So, we're leaving never. I also think that's dangerous here to use ret because someone could use this variable and break the power domain init routine.